packages/content-management/table_list/src/reducer.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React from 'react';
+import { sortBy } from 'lodash';
+import { i18n } from '@kbn/i18n';
+
+import { UpdatedAtField } from './components';
+import type { State, UserContentCommonSchema } from './table_list_view';
+import type { Action } from './actions';
+import type { Services } from './services';
+
+interface Dependencies {
+  DateFormatterComp: Services['DateFormatterComp'];
+}
+
+function onInitialItemsFetch<T extends UserContentCommonSchema>(
+  items: T[],
+  { DateFormatterComp }: Dependencies
+) {
+  // We check if the saved object have the "updatedAt" metadata
+  // to render or not that column in the table
+  const hasUpdatedAtMetadata = Boolean(items.find((item) => Boolean(item.updatedAt)));
+
+  if (hasUpdatedAtMetadata) {
+    // Add "Last update" column and sort by that column initially
+    return {
+      tableSort: {
+        field: 'updatedAt' as keyof T,
+        direction: 'desc' as const,
+      },
+      tableColumns: [
+        {
+          field: 'updatedAt',
+          name: i18n.translate('contentManagement.tableList.lastUpdatedColumnTitle', {
+            defaultMessage: 'Last updated',
+          }),
+          render: (field: string, record: { updatedAt?: string }) => (
+            <UpdatedAtField dateTime={record.updatedAt} DateFormatterComp={DateFormatterComp} />
+          ),
+          sortable: true,
+          width: '150px',
```

Comment:
I know I had asked this before, but please remind me - why do we hardcode width like this?

Reply:
We don't need a flexible column width that occupies spaces as we know it will never be longer that 150px. So we leave space for the other columns.